fritz

```diff
@@ -9,310 +8,596 @@ import requests
+            user_id=config["kowalski"]["server"]["admin_username"],
+            jwt_secret=config["kowalski"]["server"]["JWT_SECRET_KEY"],
+            jwt_algorithm=config["kowalski"]["server"]["JWT_ALGORITHM"],
+            jwt_exp_delta_seconds=int(
+                config["kowalski"]["server"]["JWT_EXP_DELTA_SECONDS"]
+            ),
+        )
+
+        config["skyportal"]["app"]["kowalski"]["token"] = kowalski_token
+
+        # get Docker gateway IP
+        command = [
             "docker",
-            "exec",
-            "-i",
-            "skyportal_web_1",
-            "/bin/bash",
-            "-c",
-            "cat /skyportal/.tokens.yaml",
-        ],
-        cwd="skyportal",
-        capture_output=True,
-        universal_newlines=True,
-    )
-    token = result.stdout.split()[-1]
+            "network",
+            "inspect",
+            "bridge",
+            '--format={{index .IPAM.Config 0 "Gateway"}}',
+        ]
+        p = subprocess.run(command, stdout=subprocess.PIPE)
+        gateway_ip = p.stdout.decode("utf-8").strip()
+        config["kowalski"]["skyportal"]["host"] = gateway_ip
+        config["skyportal"]["app"]["kowalski"]["host"] = gateway_ip
+
+        # strip down, adjust, and copy over to Kowalski and SkyPortal
+        config_kowalski = {"kowalski": config["kowalski"]}  # don't need the SP stuff
+        with open("kowalski/config.yaml", "w") as config_yaml:
+            yaml.dump(config_kowalski, config_yaml)
+
+        # Docker-specific SkyPortal stuff:
+        config["skyportal"]["database"]["host"] = "db"
+        config["skyportal"]["server"]["url"] = "http://localhost:5000"
+        config_skyportal = config["skyportal"]  # don't need the K stuff
+        with open("skyportal/config.yaml", "w") as config_yaml:
+            yaml.dump(config_skyportal, config_yaml)
+
+        # update fritz.yaml:
+        with open(c, "w") as config_yaml:
+            yaml.dump(config, config_yaml)
+
+    @staticmethod
+    def _skyportal_api(method, endpoint, token, data=None):
+        headers = {"Authorization": f"token {token}"}
+        response = requests.request(method, endpoint, json=data, headers=headers)
+        return response
+
+    @staticmethod
+    def _generate_kowalski_token(
+        user_id: str,
+        jwt_secret: str,
+        jwt_algorithm: str = "HS256",
+        jwt_exp_delta_seconds: int = 30 * 86400 * 12,
+    ):
+        """
+        Generate a token for SkyPortal to access Kowalski
+        """
+        import jwt
+
+        jwt_config = {
+            "user_id": user_id,
+            "JWT_SECRET": jwt_secret,
+            "JWT_ALGORITHM": jwt_algorithm,
+            "JWT_EXP_DELTA_SECONDS": jwt_exp_delta_seconds,
+        }
+
+        payload = {
+            "user_id": jwt_config["user_id"],
+            "exp": (
+                datetime.datetime.utcnow()
+                + datetime.timedelta(seconds=jwt_config["JWT_EXP_DELTA_SECONDS"])
+            ),
```

Comment:
Do we need the expiration date here? Maybe pin the latest K and get rid of it now that it's optional?

Reply:
Ah, thanks for catching this, made it optional.